pkg/kademlia/kademlia.go

```diff
@@ -70,16 +70,20 @@ func NewKademlia(id dht.NodeID, bootstrapNodes []pb.Node, address string, identi
 	}
 	kdb, ndb := dbs[0], dbs[1]
 
-	rt, err := NewRoutingTable(&self, kdb, ndb, &RoutingOptions{
-		idLength:     kadconfig.DefaultIDLength,
-		bucketSize:   kadconfig.DefaultBucketSize,
-		rcBucketSize: kadconfig.DefaultReplacementCacheSize,
+	rt, err := NewRoutingTable(self, kdb, ndb, &RoutingOptions{
+		IDLength:     kadconfig.DefaultIDLength,
+		BucketSize:   kadconfig.DefaultBucketSize,
+		RCBucketSize: kadconfig.DefaultReplacementCacheSize,
 	})
-
 	if err != nil {
 		return nil, BootstrapErr.Wrap(err)
 	}
 
+	return NewKademliaWithRoutingTable(self, bootstrapNodes, identity, kadconfig.Alpha, rt)
+}
+
+// NewKademliaWithRoutingTable returns a newly configured Kademlia instance
+func NewKademliaWithRoutingTable(self pb.Node, bootstrapNodes []pb.Node, identity *provider.FullIdentity, alpha int, rt *RoutingTable) (*Kademlia, error) {
```

Comment:
Why not just `NewKademlia`? every kademlia should have a routing table

Reply:
this is probably not relevant, but you can have a service be a kademlia client without a routing table. if all you're doing is lookups you can just kick off some workers against some well known nodes you have.

Reply:
This is just so I can provide a RoutingTable initialized with a inmemory database, without disrupting the existing functions.